packages/components/atoms/f-button/src/components/_tests/Button.test.js

```diff
@@ -8,4 +8,81 @@ describe('Button', () => {
         const wrapper = shallowMount(FButton, { propsData });
         expect(wrapper.exists()).toBe(true);
     });
+
+    describe('computed :: ', () => {
+        describe('buttonSizeClassname :: ', () => {
+            it('should capitalise to first letter of `buttonSize` prop :: ', () => {
+                // Arrange
+                const propsData = { buttonSize: 'medium' };
+
+                // Act
+                const wrapper = shallowMount(FButton, { propsData });
+
+                // Assert
+                expect(wrapper.vm.buttonSizeClassname).toEqual('Medium');
+            });
+
+            it('should capitalise to first two letters of `buttonSize` prop if `xsmall` :: ', () => {
+                // Arrange
+                const propsData = { buttonSize: 'xsmall' };
+
+                // Act
+                const wrapper = shallowMount(FButton, { propsData });
+
+                // Assert
+                expect(wrapper.vm.buttonSizeClassname).toEqual('XSmall');
+            });
+        });
+
+        describe('componentType :: ', () => {
+            describe('when `href` prop is an empty string :: :: ', () => {
+                const propsData = { href: '' };
+
+                it('should return `action`', () => {
```

Comment:
`Action`

Reply:
changed to `componentType` of `action-button` after chat :)